wallet/src/account/mod.rs

```diff
@@ -763,7 +766,13 @@ impl Account {
                     signed_tx.transaction().clone().into(),
                     tx_state,
                 ));
-                self.add_wallet_tx_if_relevant(db_tx, wallet_events, wallet_tx)?;
+                if self.add_wallet_tx_if_relevant(db_tx, wallet_events, wallet_tx)? {
+                    let id = AccountWalletSTxId::new(
+                        self.get_account_id(),
+                        signed_tx.transaction().get_id(),
+                    );
+                    db_tx.del_signed_transaction(&id)?;
+                }
```

Comment:
This bit of code seems to be repeated three times. Could it be factored out into a method?

Reply:
extracted